core/network/examples/peer_gossip.rs

```diff
@@ -0,0 +1,242 @@
+            tokio::time::sleep(Duration::from_secs(5)).await;
+            continue;
+        }
+
+        // Fake test data.
+        let node_id: avalanche_types::ids::node::Id = avalanche_types::ids::node::Id::from_slice(&random_manager::secure_bytes(20).unwrap());
+
+        let res_bytes = handler.app_gossip(node_id, buf[0..n].to_vec()).await.expect("Issue while attempting to gossip in fake_handler_logic");
+
+        let mut guard = client_socket.try_lock().expect("Lock of client_socket failed");
+
+        let _ = guard.write_all(&res_bytes).await;
+    }
+}
+
+async fn start_fake_node(gossip_handler_addr: String, listener_handler_addr: String, client_addr: String) {
+    // Initialize the configuration for the gossiper
+    let config = Config {
+        namespace: "test".to_string(),
+        frequency: Duration::from_secs(10),
+        poll_size: 1, // As we only have 1 other "node" in our test setup, set it to 1
+    };
+
+    // Create a TcpListener to receive messages on.
+    // Wrapping it in Arc and Mutex to safely share it between threads.
+    let handler_listener = Arc::new(Mutex::new(TcpListener::bind(listener_handler_addr.clone()).await.unwrap()));
+    let gossip_listener = Arc::new(Mutex::new(TcpListener::bind(gossip_handler_addr.clone()).await.unwrap()));
+
+    // Create a TcpStream to send messages to.
+    // Wrapping it in Arc and Mutex to safely share it between threads.
+    let stream = Arc::new(Mutex::new(TcpStream::connect(client_addr).await.unwrap()));
+
+    // Initialize the configuration for the handler and create a new handler
+    let handler_config = HandlerConfig { namespace: "test".to_string(), target_response_size: 100 };
+    let handler = new_handler(
+        handler_config,
+        Arc::new(Mutex::new(MockSet { set: Vec::<TestGossipableType>::new() })),
+    );
+
+    // Clone listener and stream for use inside the spawned task.
+    let listener_clone = handler_listener.clone();
+    let stream_clone = stream.clone();
+    // Spawn an asynchronous task that will handle incoming connections in a loop
+    tokio::spawn(async move {
+        // Accept incoming connections and spawn a new task to handle each connection
+        let (listener_socket, _) = listener_clone.try_lock().expect("Error acquiring lock on listener_clone").accept().await.unwrap();
+        fake_handler_server_logic(listener_socket, stream_clone.clone(), handler.clone()).await;
+    });
+
+    // Initialize a MockSet and populate it with some test data
+    let set = Arc::new(Mutex::new(MockSet { set: Vec::<TestGossipableType>::new() }));
+    // Generating fake data and pushing to set
+    {
+        for _ in 0..3 {
+            set.try_lock().expect("Error acquiring lock on set").set.push(
+                TestGossipableType { id: Id::from_slice(&random_manager::secure_bytes(32).unwrap()) }
+            );
+        }
+    }
+
+    {
+        assert_eq!(set.try_lock().expect("Failed to acquire lock").set.len().clone(), 3);
+    }
```

Comment:
You shouldn't need it here, either. 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=d21e8482771355894e1af0e105bad27d